src/lib.rs

```diff
@@ -27,102 +28,1090 @@ pub extern "C" fn wapc_init() {
     register_function("protocol_version", protocol_version_guest);
 }
 
+fn is_valid_id(id: i64, ranges: &[IDRange]) -> bool {
+    for range in ranges {
+        if id >= range.min && id <= range.max {
+            return true;
+        }
+    }
+    false
+}
+
+fn enforce_run_as_user_rule(
+    security_context_option: Option<apicore::SecurityContext>,
+    validation_request: &ValidationRequest<Settings>,
+) -> Result<Option<apicore::SecurityContext>, String> {
+    let mut security_context = security_context_option.unwrap_or(apicore::SecurityContext {
+        capabilities: None,
+        allow_privilege_escalation: None,
+        privileged: None,
+        proc_mount: None,
+        read_only_root_filesystem: None,
+        run_as_group: None,
+        run_as_non_root: None,
+        run_as_user: None,
+        se_linux_options: None,
+        seccomp_profile: None,
+        windows_options: None,
+    });
+    if validation_request.settings.run_as_user.rule == "MustRunAs" {
+        if security_context.run_as_user.is_some() {
+            let user_id = security_context.run_as_user.unwrap();
+            if !is_valid_id(user_id, &validation_request.settings.run_as_user.ranges) {
+                return Err("User ID outside defined ranges".to_string());
+            }
+        } else {
+            let default_user_id = validation_request.settings.run_as_user.ranges[0].min;
+            security_context.run_as_user = Some(default_user_id);
+            return Ok(Some(security_context));
+        }
+    } else if validation_request.settings.run_as_user.rule == "MustRunAsNonRoot" {
+        if security_context.run_as_non_root.is_some() && !security_context.run_as_non_root.unwrap()
+        {
+            return Err("RunAsNonRoot should be set to true".to_string());
+        }
+        if security_context.run_as_user.is_some() {
+            let user_id = security_context.run_as_user.unwrap();
+            if user_id == 0 {
+                return Err("Invalid user ID: cannot run container with root ID (0)".to_string());
+            }
+        }
+        security_context.run_as_non_root = Some(true);
+        return Ok(Some(security_context));
+    }
+    Ok(None)
+}
+
+fn enforce_run_as_group(
+    security_context_option: Option<apicore::SecurityContext>,
+    validation_request: &ValidationRequest<Settings>,
+) -> Result<Option<apicore::SecurityContext>, String> {
+    let mut security_context = security_context_option.unwrap_or(apicore::SecurityContext {
+        capabilities: None,
+        allow_privilege_escalation: None,
+        privileged: None,
+        proc_mount: None,
+        read_only_root_filesystem: None,
+        run_as_group: None,
+        run_as_non_root: None,
+        run_as_user: None,
+        se_linux_options: None,
+        seccomp_profile: None,
+        windows_options: None,
+    });
```

Comment:
I think this API's implement the `Default` trait, so you could do the following:

```suggestion
    let mut security_context = security_context_option.unwrap_or_default();
```